core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -274,49 +325,225 @@ private void maybeResetInputAndSeekToEndIntermediateTopicOffsets(final Map consu
                 }
             }
 
-        } catch (final RuntimeException e) {
+        } catch (final Exception e) {
             System.err.println("ERROR: Resetting offsets failed.");
             throw e;
         }
         System.out.println("Done.");
     }
 
-    private void maybeSeekToEnd(final KafkaConsumer<byte[], byte[]> client,
+    // visible for testing
+    public void maybeSeekToEnd(final String groupId,
+                                final Consumer<byte[], byte[]> client,
                                 final Set<TopicPartition> intermediateTopicPartitions) {
-
-        final String groupId = options.valueOf(applicationIdOption);
-        final List<String> intermediateTopics = options.valuesOf(intermediateTopicsOption);
-
         if (intermediateTopicPartitions.size() > 0) {
             System.out.println("Following intermediate topics offsets will be reset to end (for consumer group " + groupId + ")");
-            for (final String topic : intermediateTopics) {
-                if (allTopics.contains(topic)) {
-                    System.out.println("Topic: " + topic);
+            for (final TopicPartition topicPartition : intermediateTopicPartitions) {
+                if (allTopics.contains(topicPartition.topic())) {
+                    System.out.println("Topic: " + topicPartition.topic());
                 }
             }
-            if (!dryRun) {
-                client.seekToEnd(intermediateTopicPartitions);
+
+            client.seekToEnd(intermediateTopicPartitions);
```

Comment:
Should this call be governed by !dryRun

Reply:
It's fine. `seekToEnd()` is lazy and we guard `position()` with `dryRun`-check:

https://github.com/apache/kafka/pull/4159/files/fab095f46803e2f1795b4f128a607d5fdbd51423#diff-52e29464b15979ca6abaf9da60e57f86R309